tests/v2/validation/rbac/rbac_psa_test.go

```diff
@@ -291,12 +314,35 @@ func (rb *PSATestSuite) TestPSA() {
 		})
 
 		rb.Run("Testcase - Validate if members with roles "+role+"can add/edit/delete labesl from admin created namespace", func() {
-			rb.ValidatePSA(role)
+			if role == roleCustomCreateNS {
+				customRole = true
+			} else {
+				customRole = false
+			}
+			rb.ValidatePSA(role, customRole)
 		})
 
 		if strings.Contains(role, "cluster") {
 			rb.Run("Additional testcase - Validate if members with roles "+role+"can add/edit/delete labels from admin created namespace", func() {
 				rb.ValidateAdditionalPSA(role)
+				if role == roleMember {
+					err := users.AddClusterRoleToUser(rb.client, rb.cluster, rb.nonAdminUser, rb.psaRole.ID)
+					require.NoError(rb.T(), err)
+					rb.ValidateAdditionalPSA(psaRole)
+				}
+			})
+		}
+
+		if strings.Contains(role, "project") || role == roleCustomCreateNS {
+			rb.Run("Additional testcase - Validate if "+role+" with an additional role update-psa can add/edit/delete labels from admin created namespace", func() {
+				err := users.AddClusterRoleToUser(rb.client, rb.cluster, rb.nonAdminUser, rb.psaRole.ID)
+				require.NoError(rb.T(), err)
+				if role == roleCustomCreateNS {
+					customRole = true
+				} else {
```

Comment:
Same as previously commented:

Tiny nit: Optionally, consider removing else for cleaner code. By defaulting customRole to false and changing only if needed, the code could be slightly more readable.

```go
customRole = false
if role == roleCustomCreateNS {
    customRole = true
}
```

Reply:
agreed